services/web-server/src/servers/createSubscriptionServer.js

```diff
@@ -2,16 +2,55 @@ import { SubscriptionServer } from 'subscriptions-transport-ws';
 import { execute, subscribe } from 'graphql';
 import credentials from './credentials.js';
 import formatError from './formatError.js';
+import scopeUtils from 'taskcluster-lib-scopes';
+import taskcluster from 'taskcluster-client';
+import { decryptToken } from './decryptToken.js';
+import { ErrorReply } from 'taskcluster-lib-api';
 
+// TODO: Check for expiration of access token when the websocket connection is active
```

Comment:
I wonder if client does reconnect automatically in case the connection drops?
This we we could simply put a timeout and force to reauthenticate every few hours or so, would be a cheap hacky way to avoid making this code more complex

Reply:
For this I added a timeout and closed the connection. The connection was being retried but was failing to connect afterward. Still looking into this.

Reply:
what should be the ideal number of hours for the timeout to configure?

Reply:
Added the timeout in config.yml 👍

Reply:
Thanks!